### ISSUE TITLE
Easter eggs?

### ISSUE THREAD

**Issue by DiddiLeija** [DiddiLeija/diddi-and-the-bugs#13]
I'm looking for ideas and proposals of hidden features to implement here. Something that is not mentioned in the [guide](https://github.com/DiddiLeija/diddi-and-the-bugs/blob/main/guide.md), but the users can find and use!

All ideas are welcome.

cc @HarshNarayanJha

**Comment by HarshNarayanJha** [comment]
Very sorry for responding soooooooo late. I was busy due to school exams and then cancellation of it due to corona and then online classes... But now I'm back with a bugfix and also thought about some Easter eggs. Will soon open a PR.
Thanks.

**Comment by DiddiLeija** [comment]
@HarshNarayanJha and I found an easter egg accidentally! See #18.

**Comment by DiddiLeija** [comment]
Hmm... I've been thinking about the "fire trick" we made. I want to play with https://github.com/kitao/pyxel#color-palette to change the bullet color when experiencing the trick.

Here's my preview:

![Normal bullet](https://user-images.githubusercontent.com/83621221/150569697-e6e4ccc4-df09-4cf7-bba0-de53bb920ae3.png)

![Bullet trick](https://user-images.githubusercontent.com/83621221/150569751-d6a4d9a4-526e-4062-82f7-33d223ba402c.png)


I'm between using colors 9 (orange) or 3 (green, showed above).

**Comment by HarshNarayanJha** [comment]
Yeah, that's great. I will excite the player as _What, different coloured bullets. Have I discovered something unusual_
We can also make the game more interactive by giving a notification to the player from Diddi (who modified the spaceship, as in your story), that the player has unlocked a secret function of the ship....

**Comment by DiddiLeija** [comment]
Great idea!

In that case, I may set a higher width/height in the `pyxel.init`, so we can make space for text notifications. I may be working on this on the next week :wink:.

**Comment by DiddiLeija** [comment]
> Hmm... I've been thinking about the "fire trick" we made. I want to play with https://github.com/kitao/pyxel#color-palette to change the bullet color when experiencing the trick.

I already pushed this at #26.

**Comment by DiddiLeija** [comment]
I'm thinking about "secret keys" that unlock special feats. For example:

- Enemies/prizes that are not shown when the secret is not found
- A special move or function?

**Comment by HarshNarayanJha** [comment]
> I'm thinking about "secret keys" that unlock special feats. For example:
> 
> * Enemies/prizes that are not shown when the secret is not found
> * A special move or function?

Like a grand enemy type thing which yields 1000 pts or 2000 pts, or large/big trash, or we could make this grand enemy's appearing probability very less, which makes it appear only once in a single play. It will move very fast, and it will give something big like ~5000 pts. 

Or, player could use a special key combo. which can only be used once in a single play (like Z-moves, if you know pokemon...), and it will cause all the enemies/trashes on the screen to be killed, yielding very large number of points at once. This could be used when the player sees a large number of trashes on-screen...
_And we could possibly create some mind-blowing animations for this..._

More ideas on the way....

**Comment by DiddiLeija** [comment]
> Like a grand enemy type thing which yields 1000 pts or 2000 pts, or large/big trash, or we could make this grand enemy's appearing probability very less, which makes it appear only once in a single play. It will move very fast, and it will give something big like ~5000 pts.

See #41, it takes this idea.

**Comment by DiddiLeija** [comment]
It's been more than a year since our last "easter egg" proposal. Is there anything else we could add? If not, I think we _could_ close this issue.

**Comment by HarshNarayanJha** [comment]
Yes, go ahead!

**Comment by DiddiLeija** [comment]
Ok!

I think want to resurrect this idea:

> Or, player could use a special key combo. which can only be used once in a single play (like Z-moves, if you know pokemon...), and it will cause all the enemies/trashes on the screen to be killed, yielding very large number of points at once. This could be used when the player sees a large number of trashes on-screen...

Sounds like something possible!

**Comment by HarshNarayanJha** [comment]
Logic shall be simple enough. I guess I can at least code it on my iPad

**Comment by DiddiLeija** [comment]
Ok, if you wanna give it a try, I'll separate this on another issue, and I'll assign you as a reminder ;)
